mods/boats/init.lua

```diff
@@ -180,15 +171,13 @@ function boat.on_step(self, dtime)
 		self.object:set_pos(self.object:get_pos())
 		return
 	end
-	local s = get_sign(self.v)
-	self.v = self.v - dtime * 0.6 * s
-	if s ~= get_sign(self.v) then
-		self.object:set_velocity({x = 0, y = 0, z = 0})
+	-- We need to preserve velocity sign to properly apply drag force while moving backward
+	local drag = dtime * math.sign(self.v) * (0.01 + 0.08 * self.v * self.v)
```

Comment:
The 0.08 needs altering slightly so that max velocity remains exactly 5n/s.
Seems to need to be:
1.99/25 = 0.0796

Reply:
Updated